pkg/controller/datavolume-controller.go

```diff
@@ -347,6 +347,24 @@ func (r *DatavolumeReconciler) Reconcile(_ context.Context, req reconcile.Reques
 		return reconcile.Result{}, nil
 	}
 
+	// If sourceRef is set, populate spec.Source with data from the DataSource
+	if datavolume.Spec.SourceRef != nil {
```

Comment:
would it be better to extract this to a function?  That reconcile function is already big may be more readable

Reply:
sure